x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/CacheDirectory.java

```diff
@@ -287,23 +293,21 @@ int readCacheFile(FileChannel fc, long end, long position, byte[] buffer, int of
         @SuppressForbidden(reason = "Use positional writes on purpose")
         void writeCacheFile(FileChannel fc, long start, long end) throws IOException {
             assert assertFileChannelOpen(fc);
-            final byte[] copyBuffer = new byte[Math.toIntExact(Math.min(COPY_BUFFER_SIZE, end - start))];
+            final long length = end - start;
+            final byte[] copyBuffer = new byte[Math.toIntExact(Math.min(COPY_BUFFER_SIZE, length))];
             logger.trace(() -> new ParameterizedMessage("writing range [{}-{}] to cache file [{}]", start, end, cacheFileReference));
 
             int bytesCopied = 0;
-            try (IndexInput input = in.openInput(cacheFileReference.getFileName(), ioContext)) {
+            final long startTimeNanos = currentTimeNanosSupplier.getAsLong();
+            try (InputStream input = openInputStream(start, length)) {
```

Comment:
There is an important change here: by reading the range to cache on disk using the `BlobContainer` only the required bytes are downloaded in a (usually) single request (unless the range spans over multiple blob parts) and there is no read ahead engaged here.

Reply:
:+1: